### DISCUSSION TITLE
no avx2版本可以直接升级为avx2版本吗？

### DISCUSSION THREAD

**Discussion by MuXueMaster** [apache/doris#34746] [A - General / Q&A]
我们的服务器是支持  avx2的指令，但是我安装选择的版本是  doris 2.0.2 no avx2 ， 之所以选择no avx2，是想着这个的兼容性比 avx2更好一些（既可以在支持avx2指令的服务器上安装，也可以在不支持avx2指令的服务器上安装），那么其稳定性肯定比  avx2更好一些。

运行了大概2-3个月,数据量多大几十亿条，领导要求 换成 doris 2.0.3 avx2（理由是 性能更高），  我想咨询下，doris 2.0.2 no avx2 可以直接升级为 doris 2.0.3 avx2吗？ 升级中是否有特别注意事项？ 对业务数据是否会造成影响？会有什么影响？
升级的方法是否跟官网提供的一样？
 https://doris.apache.org/zh-CN/docs/admin-manual/cluster-management/upgrade  

（一直想不明白，doris为啥要区分avx2与no avx2的版本，为啥不在代码里自动区分，或者配置文件中加个字段区分？）

**Comment by yagagagaga** [comment]
可以，但可能有兼容性问题，建议现在测试环境充分测试验证，推荐使用[doris-manager](https://doris.apache.org/ecosystem/cluster-management)升级。

**Comment by zclllyybb** [comment]
因为avx2的是否启用，是在编译期条件编译决定的。如果在运行期决定，那么就会有不可避免的跳转开销和I-cache开销。
avx2和no-avx2的稳定性理论上没有区别。

**Reply by crazythinking** [reply]
@yagagagaga 也想了解下doris-manager是商业版本吗，如何收费呢

**Reply by yagagagaga** [reply]
可以加个微信私聊 https://github.com/apache/doris/discussions/16887

**Comment by maruyue** [comment]
可以直接升级的，没有任何问题的
